Editor/AGS.Editor/Components/AudioComponent.cs

```diff
@@ -602,6 +602,35 @@ private void CommandForceReimportOfAllAudioClips()
             Factory.GUIController.ShowOutputPanel(errors);
         }
 
+        public void ReplaceAudioClipSource(AudioClip clip)
+        {
+            string dir = Path.GetDirectoryName(clip.SourceFileName);
+            if (!Directory.Exists(dir)) dir = Factory.AGSEditor.CurrentGame.DirectoryPath;
+            string selectedFile = _guiController.ShowOpenFileDialog("Select audio to add", AUDIO_FILES_FILTER, dir);
+            if (string.IsNullOrEmpty(selectedFile) || !File.Exists(selectedFile)) return;
+
+            string fileExtension = Path.GetExtension(selectedFile).ToLower();
+            if (!_fileTypeMappings.ContainsKey(fileExtension)) return;

```

Comment:
This and above failure cases should also report a visible error, preferably with a reason. Right now if I select an unsupported file, the operation fails without explanation.